types/webpack/index.d.ts

```diff
@@ -1832,7 +1833,22 @@ declare namespace webpack {
     }
 
     class ProgressPlugin extends Plugin {
-        constructor(options?: (percentage: number, msg: string, moduleProgress?: string, activeModules?: string, moduleName?: string) => void);
+        constructor(options?: ProgressPlugin.Options | ProgressPlugin.Handler);
+    }
+
+    namespace ProgressPlugin {
+        type Handler = (percentage: number, msg: string, moduleProgress?: string, activeModules?: string, moduleName?: string) => void;
+        interface Options {
+            activeModules?: boolean;
+            entries?: boolean;
+            handler?: Handler;
+            modules?: boolean;
+            modulesCount?: number;
+            profile?: boolean;
+            dependencies?: boolean;
+            dependenciesCount?: number;
+            percentBy?: "entries" | "dependencies" | "modules" | null;
+        }
```

Comment:
```suggestion
            profile?: true | false | null;
        }
```
Two points here:

1. The official declaration says that `profile` could also be `null`:
  https://github.com/webpack/webpack/blob/v4.42.1/declarations/plugins/ProgressPlugin.d.ts#L41
2. The properties `dependencies`, `dependenciesCount` and `percentBy` seem to only exist in in the `v5.0.0-beta` but not in the current stable `v4`. Since the types here are currently für the latest stable version they should not included here.
  See https://github.com/webpack/webpack/blob/v4.42.1/declarations/plugins/ProgressPlugin.d.ts (`v4.42.1`) vs
  https://github.com/webpack/webpack/blob/v5.0.0-beta.14/declarations/plugins/ProgressPlugin.d.ts (`v5.0.0-beta.14`)
  If you need this properties anyway you can add them to the [`next.d.ts`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/webpack/next.d.ts) file where declarations for the upcoming `v5` are located.

Reply:
Thanks for your suggestion. 
I have updated the code, and also added some comments (even more detailed than webpack official code) for ProgressPlugin option.